libraries/llvm/rts.ll

```diff
@@ -56,37 +56,21 @@
 ; Pointers for a heap allocated stack
 %Mem = type { %Sp, %Base, %Limit }
 
-; This is part of a linked list containing all states in a region.
+; The garbage collector differentiates three groups of types:
+; - Values (Integer, Double)
+; - Objects (Positive, Negative)
+; - Strings
+; For each group we have an arena where mutable state is allocated.
 ;
-;   +[ StateHeader ]+--------------+
-;   | Next  | Size  | Payload ...  |
-;   +---------------+--------------+
-%State = type ptr
+%Region = type [ 3 x %Mem ]
```

Comment:
I don't quite like this solution and would prefer storing a sharer and an eraser with each object. But ok.

Reply:
This would become obsolete if we switch to monomorphic regions.